Assets/MixedRealityToolkit/Utilities/AssetScriptReferenceRetargeter.cs

```diff
@@ -0,0 +1,593 @@
+                }
+            }
+
+            DirectoryInfo[] directories = new DirectoryInfo(Application.dataPath.Replace("Assets", "NuGet/Plugins")).GetDirectories("*", SearchOption.AllDirectories);
+            string dllGuid = string.Empty;
+            bool isUAP = false;
+            foreach (DirectoryInfo directory in directories)
+            {
+                isUAP = false;
+                if (!directory.Name.Equals("EditorPlayer"))
+                {
+                    if (directory.Name.Contains("Standalone") || directory.Parent.Name.Contains("Standalone"))
+                    {
+                        metaFileContent = File.ReadAllLines(standaloneMetaFile.FullName);
+                    }
+                    else if (directory.Name.Contains("UAP") || directory.Parent.Name.Contains("UAP"))
+                    {
+                        metaFileContent = File.ReadAllLines(uapMetaFile.FullName);
+                        isUAP = true;
+                    }
+
+                    FileInfo[] files = directory.GetFiles("*.dll", SearchOption.TopDirectoryOnly);
+                    string metaFilePath = string.Empty;
+                    foreach (FileInfo file in files)
+                    {
+                        //Editor is guid + 1
+                        //Standalone is guid + 2
+                        //UAP is guid + 3
+                        dllGuid = dllGuids[file.Name];
+                        dllGuid = CycleGuidForward(dllGuid);
+                        if (isUAP)
+                        {
+                            dllGuid = CycleGuidForward(dllGuid);
+                        }
+
+                        if (dllGuids.ContainsKey(file.Name))
+                        {
+                            metaFilePath = $"{file.FullName}.meta";
+                            metaFileContent[1] = $"guid: {dllGuid}";
+                            if (File.Exists(metaFilePath))
+                            {
+                                File.Delete(metaFilePath);
+                            }
+                            File.WriteAllLines(metaFilePath, metaFileContent);
+                        }
+                    }
+                }
+            }
+        }
+
+        private static StringBuilder guidBuilder = new StringBuilder(32);
+        private static string CycleGuidForward(string guid)
+        {
+            guidBuilder.Clear();
+            //Add one to each hexit in the guid to make it unique, but also reproducible
+            foreach (char hexit in guid)
+            {
+                switch (hexit)
+                {
+                    case 'f':
+                        guidBuilder.Append('0');
+                        break;
+                    case '9':
+                        guidBuilder.Append('a');
+                        break;
+                    default:
+                        guidBuilder.Append((char)(hexit + 1));
+                        break;
+                }
+            }
```

Comment:
Do you know if we can depend on the guids in the meta file to always be lowercase? If there's a uppercase F I think we'll just go to G.

Reply:
I'll make it lower for here.

---
In reply to: [277828536](https://github.com/Microsoft/MixedRealityToolkit-Unity/pull/4051#discussion_r277828536) [](ancestors = 277828536)